includes/import-functions.php

```diff
@@ -132,58 +131,66 @@ function edd_import_coupon_csv_file() {
 						$validation_error = true;
 						$validation_msg  .= __('Discount Name is blank.', 'edd') . "; ";
 					}
-					
+
 					// discount code  blank
 					if ( 0 == strlen( $edd_discount_code ) ) {
 						$validation_error = true;
 						$validation_msg  .= __('Discount code is blank.', 'edd') . "; ";
 					}
-					
+
+					// Verify only accepted characters.
+					$sanitized_discount_code = preg_replace( '/[^a-zA-Z0-9-_]+/', '', $edd_discount_code );
+					if ( strtoupper( $edd_discount_code ) !== strtoupper( $sanitized_discount_code ) ) {
+						$validation_error = true;
+						$validation_msg  .= __('Discount code is invalid.', 'edd') . "; ";
+					}
+
 					// discount type is blank
 					if ( 0 == strlen( $edd_discount_type ) ) {
 						$validation_error = true;
 						$validation_msg  .= __('Discount type is blank.', 'edd') . "; ";
 					}
-					
-					// discount type is invaid
-					if ( ( $edd_discount_type != 'flat' ) && ( $edd_discount_type != 'percentage' ) ) {
+
+					// discount type is invalid
+					$discount_types = array( 'flat', 'percentage' );
+					if ( ! in_array( $edd_discount_type, $discount_types, true ) ) {
 						$validation_error = true;
 						$validation_msg  .= __('Discount type is invalid.', 'edd') . "; ";
 					}
-					
+
 					// discount code is blank
 					if ( 0 == strlen( $edd_discount_code ) ) {
 						$validation_error = true;
 						$validation_msg  .= __('Discount code is blank.', 'edd') . "; ";
 					}
-					
+
 					// start time validation
 					if ( ! strtotime( $edd_discount_start ) ) {
 						$validation_error = true;
 						$validation_msg  .= sprintf(__('Date validation failed for string: %s', 'edd'), $edd_discount_start) . "; ";
 					}
-					
+
 					// end time validation
 					if ( ! strtotime( $edd_discount_end ) ) {
 						$validation_error = true;
 						$validation_msg  .= sprintf(__('Date validation failed for string: %s', 'edd'), $edd_discount_end) . "; ";
 					}
-					
-					// set start time to correct date type
-					$edd_discount_start =  date( 'm/d/Y H:i:s', strtotime( $edd_discount_start ) );
-					
-					// set end time to the correct date type
-					$edd_discount_end =  date( 'm/d/Y H:i:s',  strtotime(  date( 'm/d/Y', strtotime( $edd_discount_end ) ) . ' 23:59:59' )  );
-				
-					// check status 
+
+					// check status
 					$status = array( 'active', 'inactive', 'expired' );
-					if ( ! in_array( $edd_discount_status, $status ) ) {
+					if ( ! in_array( $edd_discount_status, $status, true ) ) {
 						$validation_error = true;
 						$validation_msg  .= __('Status needs to be active, inactive, or pending', 'edd') . "; ";
 					}
-					
-					// maybe more validation later
 
+					// Check product condition.
+					$product_conditions = array( 'all', 'any' );
+					if ( ! in_array( $edd_discount_product_condition, $product_conditions, true ) ) {
+						$validation_error = true;
+						$validation_msg  .= __('Product condition is invalid.', 'edd') . "; ";
```

Comment:
```suggestion
						$validation_msg  .= __('Product condition is invalid.', 'edd');
```
Otherwise this ends with a full stop followed by a semicolon...I see that some of the other validation messages have the same issue but we can fix those when addressing the text domains. Just suggesting here as it's a new string.

Reply:
I think that this shouldn't be changed in this PR, and that should be changed when `$validation_msg` as whole is change. It follows the same pattern that others are using: error, followed by semicolon **and** space. Semicolon and space are used to separate multiple errors that will be on the same line. 

I believe that the way validation message string is generated and displayed needs full rewrite, but I think that work on that is out of scope of this PR.